### ISSUE TITLE
Rebuild Download History shortcode output

### ISSUE THREAD

**Issue by SeanTOSCD** [awesomemotive/easy-digital-downloads#6448]
This shortcode has been somewhat neglected since we have wanted people to use the `[purchase_history]` shortcode instead. However, people still like the `[download_history]` shortcode and it will remain available for use. So let's make it good.

I'll start a list of needed bug fixes/enhancements:

- [ ] Bundle purchases give access to all child product files, ignoring configuration details. See: https://github.com/easydigitaldownloads/easy-digital-downloads/issues/5894#issuecomment-374789025
- [ ] When a payment has a status of Renewal, file download links do not show up in the shortcode because of this conditional: https://github.com/easydigitaldownloads/easy-digital-downloads/blob/master/templates/history-downloads.php#L55 It should instead check in some way to see if the user has access to the file instead of depending on the status of the payment. Related ticket: https://secure.helpscout.net/conversation/676839815/91379?folderId=1448602

-----

Adjust list as necessary.

**Comment by robincornett** [comment]
Related: [[download_history] should show an indication of expired license keys #1193](https://github.com/easydigitaldownloads/EDD-Software-Licensing/issues/1193)

Software Licensing request to show if a license is expired, or prevent download links from showing if a license is expired.

**Comment by dgoldak** [comment]
download_history should not list products with no files available to download. Or have a parameter to do that if there is value in listing all downloads. For example:
<img width="712" alt="Screen Shot 2021-02-17 at 11 11 21 AM" src="https://user-images.githubusercontent.com/30910807/108241024-052ab600-7111-11eb-8b92-81c2785e6989.png">

The shortcode is useful if you want to list all the files for download in one place. In that use case, "no downloadable files found" doesn't look good.

**Comment by ashleyfae** [comment]
Related ticket: https://secure.helpscout.net/conversation/1559350278/188585/

This customer's ticket and screenshot really clearly illustrates some problems with the shortcode right now. The page lists 10 renewal payments, none of which show available downloads ("Payment status is renewal.") The one available download is buried under that list of renewals. It makes it really hard to actually find the download links.

And from the customer:

> What is the logic behind showing a payment history under a point called download history?

In short: 

1. We should be showing unique download links. (No sense listing the same product twice.)
2. We shouldn't show products that don't have downloads availble.

**Comment by mihaijoldis** [comment]
Based on https://secure.helpscout.net/conversation/1720026211/1686834?folderId=5762562 I want to bump the 2) that Ashley said.
If a download has no files attached then don't list it there

**Comment by cklosowski** [comment]
I'm closing this issue, as we're going to rebuild the download history concept into a block, and essentially leave the legacy shortcodes as they are. We'll improve the markup and make it more modern in the new block.
